_posts/2023-11-14-credential-stores-on-client-side.adoc

```diff
@@ -0,0 +1,155 @@
+---
+layout: post
+title: 'Using Credential Stores on the Elytron-Tool'
+date: 2023-11-14
+tags: credential-store elytron-tool wildfly-config ejb
+synopsis: How to set up a credential store using the elytron-tool and use it for the wildfly-config for an application deployed on WildFly. 
+author: prarthonapaul
+---
+
+:toc: macro
+:toc-title:
+
+WildFly allows the use of credential stores to keep alias for sensitive information, such as, passwords for external services. Credential stores can be used to store different credentials under aliases and use credential-reference to specify them in server configuration. As a result, the credential is no longer visible in clear text. However, there are additional uses for credential store as well. This guides explains how to create credential stores using the elytron-tool and use it for an application deployed to WildFly. 
+
+toc::[]
+
+== Prerequisite
+To follow along with this guide you will need:
+
+* about 10 minutes
+* WildFly with credential-store support  
+* Elytron-tool for WIldFly server
+
+== Access the Example Application
+For this guide we will be using the `ejb-security` example application. Let us first clone the project and navigate to it: 
+
+```
+git clone https://github.com/wildfly-security-incubator/elytron-examples
+cd ejb-security
+```
+
+== About WildFly Client Configuration
+EJBs, also known as Enterprise JavaBeans are a collection of specifications that are used for building java applications and offer a set og APIs for developing anf running secured applications. When invoking ejbs using the WildFly server, we need to configure the WildFly client to specify revevant information about authentication to secure the application. This can be done using a file named wildfly-config.xml located inside the sec/main/resources folder of the application. 
```

Comment:
s/anf running/and running

Reply:
s/ejbs/EJBs
s/revevant/relevant
s/sec/src

Reply:
Fixed!